Adaptors/Amqp/src/PushQueueStorage.cs

```diff
@@ -96,37 +94,43 @@ public class PushQueueStorage : QueueStorage, IPushQueueStorage
                      internalPriority);
 
 
-    await messages.ParallelForEach(new ParallelTaskOptions(cancellationToken),
+    await messages.ParallelForEach(new ParallelTaskOptions(parallelismLimit_,
+                                                           cancellationToken),
                                    async msgData =>
                                    {
                                      for (var retry = 0; retry < Options.MaxRetries; retry++)
                                      {
                                        try
                                        {
-                                         await using var session = await sessionPool_.GetAsync(cancellationToken)
-                                                                                     .ConfigureAwait(false);
-
-                                         var sender = new SenderLink(session,
-                                                                     Guid.NewGuid()
-                                                                         .ToString(),
-                                                                     $"{partitionId}###q{whichQueue}");
-
-                                         await sender.SendAsync(new Message(Encoding.UTF8.GetBytes(msgData.TaskId))
-                                                                {
-                                                                  Header = new Header
-                                                                           {
-                                                                             Priority = (byte)internalPriority,
-                                                                           },
-                                                                  Properties = new Properties
-                                                                               {
-                                                                                 MessageId = Guid.NewGuid()
-                                                                                                 .ToString(),
-                                                                               },
-                                                                })
-                                                     .ConfigureAwait(false);
+                                         var address = $"{partitionId}###q{whichQueue}";
+
+                                         var pool = senders_.GetOrAdd(address,
+                                                                      s => new ObjectPool<SenderLink>(parallelismLimit_,
+                                                                                                      async token => new SenderLink(new Session(await ConnectionAmqp
```

Comment:
So each Link have its own Session. Is it something we want to keep?

Reply:
for now yes